src/impl/EspNowPreferences.cpp

```diff
@@ -30,6 +31,43 @@ void EspNowPreferences::initalizeNVS() {
   }
 }
 
+bool EspNowPreferences::espNowSetChannelForHost(uint8_t channel) {
+  auto key = NVS_STORAGE_KEY_HOST_CHAN;
+  esp_err_t err = nvs_set_blob(_nvs_handle, key, &channel, 1);
+  if (err != ESP_OK) {
+    ESP_LOGE(TAG, "Failed to set blob to NVS with key %s (%s)", key, esp_err_to_name(err));
+  }
+  return err == ESP_OK;
+}
+
+bool EspNowPreferences::espNowGetChannelForHost(uint8_t *buffer) {
+  auto key = NVS_STORAGE_KEY_HOST_CHAN;
+  if (buffer == nullptr) {
+    ESP_LOGE(TAG, "channel buffer is null");
+    return false;
+  }
+
+  size_t required_size;
```

Comment:
Use `nvs_get_u8` instead. Then you don't need to reserve any size.
Example:
```
  auto key = NVS_STORAGE_KEY_HOST_CHAN;
  esp_err_t err = nvs_get_u8(_nvs_handle, key.c_str(), &buffer);
  if (err != ESP_OK) {
    ESP_LOGE(TAG, "Failed to get u8 from NVS with key %s (%s)", key.c_str(), esp_err_to_name(err));
    return false;
  }
  return true;
```

Reply:
fixed